cmd/dlt/commands/sync.go

```diff
@@ -57,32 +57,78 @@ var syncCmd = &cobra.Command{
 		if all {
 			connections = *allConnections
 		} else {
-			if connection == "" {
-				logger.Error().Msg("either --connections or --all is required")
-				return
-			}
-			connections = strings.Split(connection, ",")
-
-			for _, c := range connections {
-				if !utils.Contains(*allConnections, c) {
-					logger.Error().Msg(fmt.Sprintf("connection %v not found", c))
-					return
+			connectionsSlice := strings.Split(connectionName, ",")
+		ConnectionSelect:
+			for _, c := range connectionsSlice {
+				for _, conn := range *allConnections {
+					if strings.TrimSpace(c) == conn.Name {
+						connections = append(connections, conn)
+						continue ConnectionSelect
+					}
 				}
+				logger.Error().Msg(fmt.Sprintf("connectionName %v not found", c))
+				return
 			}
 		}
 
-		sleepDuration := time.Duration(interval * float64(time.Hour))
-
 		// Required for graceful shutdown
 		ctx, cancel := context.WithCancel(context.Background())
 		defer cancel()
 
 		shutdownChannel := make(chan os.Signal, 1)
 		signal.Notify(shutdownChannel, syscall.SIGINT, syscall.SIGTERM)
 
-		running := true
+		running := false
 		sigCount := 0
 
+		cronScheduler, err := gocron.NewScheduler()
+		if err != nil {
+			logger.Error().Str("err", err.Error()).Msg("failed to create cron scheduler")
+			return
+		}
+
+		for _, c := range connections {
+			loader, err := l.SetupLoader(configPath, c.Name, false, fromBundleId, math.MaxInt64, force)
```

Comment:
Could it be dangerous that we reuse the same loader again for every iteration? e.g. somebody adds a state variable to the loader one day, which does not properly reset after the run is finished

Reply:
A new loader is created every iteration in order to add it to the `gocron.NewTask`. Right now, this is not dangerous, but I will add a comment to keep it in mind for potential future changes.